src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -1577,25 +1577,288 @@ class QuantizedOr(QuantizedOpUnivariateOfEncrypted, QuantizedOp):
-    """Div operator /.
+class QuantizedDiv(QuantizedMixingOp):
+    """Quantized Division operator.
 
-    This operation is not really working as a quantized operation. It just works when things got
-    fused, as in e.g., Act(x) = 1000 / (x + 42))
+    Can divide either two variables (both encrypted) or a variable and a constant
     """
 
     _impl_for_op_named: str = "Div"
 
+    def __init__(
+        self,
+        *args,
+        rounding_threshold_bits: Union[None, int, Dict[str, Union[str, int]]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, rounding_threshold_bits=rounding_threshold_bits, **kwargs)
+        self.divider_quantizer: UniformQuantizer
+        self.min_non_zero_value: numpy.float64
+
+    def calibrate(self, *inputs: numpy.ndarray) -> numpy.ndarray:
+        """Create corresponding QuantizedArray for the output of the activation function.
+
+        Args:
+            *inputs (numpy.ndarray): Calibration sample inputs.
+
+        Returns:
+            numpy.ndarray: the output values for the provided calibration samples.
+        """
+
+        # If two inputs, compute the divider
+        if not self.can_fuse() and len(inputs) == 2:
+            min_non_zero_value = numpy.min(numpy.abs(inputs[1]))
+
+            # mypy
+            assert min_non_zero_value is not None and min_non_zero_value > 0
+            self.min_non_zero_value = min_non_zero_value
+
+            q_array_divider = QuantizedArray(self.n_bits, 1 / inputs[1])
+
+            # Store the quantizer of the divider
+            self.divider_quantizer = q_array_divider.quantizer
+
+        return super().calibrate(*inputs)
+
+    def q_impl(
+        self,
+        *q_inputs: ONNXOpInputOutputType,
+        calibrate_rounding: bool = False,
+        **attrs,
+    ) -> ONNXOpInputOutputType:
+
+        # If the op can be fused or if any input
+        # is a RawOpOutput, perform the op in the clear
+        if (
+            len(q_inputs) == 1
+            or any(isinstance(input, RawOpOutput) for input in q_inputs)
+            or self.can_fuse()
+        ):
+            return super().q_impl(*q_inputs, **attrs)
+
+        # For mypy
+        assert self.output_quant_params is not None
+        assert self.output_quant_params.scale is not None
+        assert self.output_quant_params.zero_point is not None
+
+        prepared_inputs = self._prepare_inputs_with_constants(
+            *q_inputs, calibrate=False, quantize_actual_values=True
+        )
+
+        q_input_0: QuantizedArray = prepared_inputs[0]
+        q_input_1: QuantizedArray = prepared_inputs[1]
+
+        assert q_input_0.quantizer.scale is not None
+        assert q_input_0.quantizer.zero_point is not None
+
+        assert q_input_1.quantizer.scale is not None
+        assert q_input_1.quantizer.zero_point is not None
+
+        # Dequantize
+        input_1 = q_input_1.dequant()
+
+        # Replace input_1 with min_non_zero_qvalue if input_1 is 0
```

Comment:
`min_non_zero_qvalue`should be `min_non_zero_value`

Reply:
@RomanBredehoft very well spotted!